flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/catalog/CatalogCalciteSchema.java

```diff
@@ -57,16 +62,34 @@ public CatalogCalciteSchema(String catalogName, Catalog catalog, boolean isStrea
 	 */
 	@Override
 	public Schema getSubSchema(String schemaName) {
-		if (catalog.databaseExists(schemaName)) {
-			return new DatabaseCalciteSchema(schemaName, catalogName, catalog, isStreamingMode);
+		if (temporaryDatabaseExists(schemaName) || permanentDatabaseExists(schemaName)) {
+			return new DatabaseCalciteSchema(schemaName, catalogName, catalogManager, isStreamingMode);
 		} else {
 			return null;
 		}
 	}
 
+	private boolean temporaryDatabaseExists(String schemaName) {
+		return catalogManager.getTemporaryTables()
+			.keySet()
+			.stream()
+			.anyMatch(i -> i.getDatabaseName().equals(schemaName));
```

Comment:
I'm wondering if we should move this logic into `CatalogManager`. Because once we start case insensitivity, those `name.equals(otherName)` checks are hard to find.